addons/payment_stripe/models/payment.py

```diff
@@ -355,7 +359,10 @@ def _create_stripe_refund(self):
         }
 
         _logger.info('_create_stripe_refund: Sending values to stripe URL, values:\n%s', pprint.pformat(refund_params))
-        res = self.acquirer_id._stripe_request('refunds', refund_params)
+        # Create an idempotency key using the hash of the transaction reference and the database UUID
+        database_uuid = self.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        idempotency_key = sha1((database_uuid + self.reference).encode("utf-8")).hexdigest()
```

Comment:
@mafo-odoo I played a bit with Stripe's idempotency keys and it turns out that they *do not* allow the same key for two different endpoints!
![image](https://user-images.githubusercontent.com/15896813/196439290-b9b948fc-0016-4a0e-9658-1daff82f3cf3.png)

This means that 14.0 needs a fix, which could be as simple as this:
```suggestion
        idempotency_key = sha1((database_uuid + self.reference + 'refunds').encode("utf-8")).hexdigest()
```

I will adapt the code in the forward port PR for 15.0 so that the fix is only needed in 14.0.